src/topp/IDFileConverter.cpp

```diff
@@ -615,28 +618,67 @@ class TOPPIDFileConverter :
-      for (Size i = 0; i < peptide_identifications.size(); ++i)
+      Int max_hits = getIntOption_("number_of_hits");
+      if (max_hits == 0)
+      {
+        OPENMS_LOG_WARN << "'number of hits' set to 0. No peptide hits will be exported! Please enter the number of peptide hits you wish to export. To export all hits enter a negativ number." << endl;
+      }
+      if (max_hits < 0)
+      {
+        max_hits = INT_MAX;
+      }
+
+      bool concat = getFlag_("concatenate_peptides");
+      //Because by concatenation of peptides [KR]|P sites will probably be created, peptides starting with 'P' are
+      //saved separatly and later moved to the beginning of the concatenated sequence.
+      //This is done to avoid losing information about the preceding peptides if a peptides starts with 'P'.
+      String all_p; //peptides beginning with 'P'
+      String all_but_p; //all the others
+
+      FASTAFile f;
+      f.writeStart(out);
+      FASTAFile::FASTAEntry entry;
+      for (const PeptideIdentification& pep_id : peptide_identifications)
       {
-        for (Size l = 0; l < peptide_identifications[i].getHits().size(); ++l)
+        Int curr_hit = 1;
+        for (const PeptideHit& hit : pep_id.getHits())
         {
-          const PeptideHit& hit = peptide_identifications[i].getHits()[l];
-          String seq = hit.getSequence().toUnmodifiedString();
-          std::set<String> prot = hit.extractProteinAccessionsSet();
-          fasta << ">" << seq
-                << " " << ++count
-                << " " << hit.getSequence().toString()
-                << " " << ListUtils::concatenate(StringList(prot.begin(), prot.end()), ";")
-                << "\n";
-          // FASTA files should have at most 60 characters of sequence info per line
-          for (Size j = 0; j < seq.size(); j += 60)
+          if (curr_hit > max_hits) break;
+          ++curr_hit;
+
+          if (concat)
+          {
+            String seq = hit.getSequence().toUnmodifiedString();
+            if (seq[0] == 'P')
+            {
+              all_p += seq;
+            }
+            else
+            {
+              all_but_p += seq;
+            }
+          }
+          else
           {
-            Size k = min(j + 60, seq.size());
-            fasta << seq.substr(j, k - j) << "\n";
+            String seq = hit.getSequence().toUnmodifiedString();
+            std::set<String> prot = hit.extractProteinAccessionsSet();
+            entry.sequence = seq;
+            entry.identifier = seq;
+            entry.description = String(count) + " " + hit.getSequence().toString() + " " + ListUtils::concatenate(StringList(prot.begin(), prot.end()), ";");
+
+            f.writeNext(entry);
+            ++count;
           }
         }
       }
-      fasta.close();
+      if (concat)
+      {
+        entry.sequence = all_p + all_but_p;
+        entry.identifier = protein_identifications[0].getSearchEngine();
+        entry.description = "concatenated peptides";
```

Comment:
Maybe take this description as ´const char* CONCAT_PEPTIDE = "concatenated peptide";´ which you place into the library and use from all places you need it (see Constants.h?)

Reply:
Your thought is that this can later be used again for searching for those peptides in the db suitability tool, right?
This wouldn't work though, because the description is lost during the peptide identification search, isn't it?
Or am I missing something here?

Reply:
very good thinking! :)
prepend it to the identifier then?

Reply:
(omit spaces and other funny stuff, use underscores if required)